### PR TITLE
Upgrade React to version 16

### LINKED ISSUE #1978: Upgrade to React 16
With https://github.com/kolide/fleet/pull/1976 we have fixed the deprecation warnings for React APIs we were using in 15.x that are gone in 16.x. We should do whatever further refactoring we can to get on a more recent version.

### PR DESCRIPTION
**Pull request by zwass**
- Update all associated dependencies
- Very minimal changes to components
- Extensive refactoring for broken tests

Closes #1978

### CODE REVIEW

File: frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx [hunk 1]
```diff
@@ -76,15 +76,17 @@ describe('QueryForm - component', () => {
...
+    let saveChangesOption = form.find('li.dropdown-button__option').first().find('Button');
 
     expect(saveChangesOption.props()).toInclude({
       disabled: true,
     });
 
     fillInFormInput(nameInput, 'New query name');
+    nameInput.simulate('change', { target: { value: 'New query name' } });
 
+    saveChangesOption = form.find('li.dropdown-button__option').first().find('Button');
```
Comment (kyleknighted) [inline frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx:89]: Why is it necessary to reassign the same selector?

File: frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx [hunk 2]
```diff
@@ -94,14 +96,15 @@ describe('QueryForm - component', () => {
...
-    const saveChangesOption = form.find('li.dropdown-button__option').first().find('Button');
+    let saveChangesOption = form.find('li.dropdown-button__option').first().find('Button');
 
     expect(saveChangesOption.props()).toInclude({
       disabled: true,
     });
 
     fillInFormInput(descriptionInput, 'New query description');
 
+    saveChangesOption = form.find('li.dropdown-button__option').first().find('Button');
```
Comment (kyleknighted) [inline frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx:107]: Same as above

File: frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx (hunk 1 above)
Comment (zwass) [reply frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx:89]: The components returned from the selector seem to now be immutable in the new version of Enzyme... Only the root element is "rerendered", so you have to find them again from that root element in order to capture the changes.

Comment (kyleknighted) [reply frontend/components/forms/queries/QueryForm/QueryForm.tests.jsx:89]: That seems very annoying

**Comment by zwass** [comment]
I gave a full run through the app from setup, adding hosts, live query, editing queries, packs, and settings. I think everything is intact after the upgrade.

**Pull request merged**